and-cli-deploy-jenkins.js

```diff
@@ -0,0 +1,184 @@
+            const userPrompt = prompt.getPrompt();
+
+            file.deleteIfExists(configPath);
+            shell.touch(configPath);
+
+            const url = await userPrompt.questionAsync("Jenkins server url: ");
+            const username = await userPrompt.questionAsync(
+                "Jenkins Username: "
+            );
+            const token = await userPrompt.questionAsync("Jenkins API Token: ");
+
+            let config = jenkins.getConfig();
+            config = Object.assign(config, { url, username, token });
+
+            const configResult = jenkins.writeToConfig(config);
+            if (!configResult) {
+                echo.error("Error writing to config file");
+                shell.exit(1);
+            }
+            await prompt.confirmOrExit("Success, add job profile?", 0);
+            await this.createProfile();
+        },
+
+        async createProfile() {
+            const userPrompt = prompt.getPrompt();
+            const jenkinsJobName = await userPrompt.questionAsync(
+                "Jenkins Job Name: "
+            );
+            const profileName = await userPrompt.questionAsync(
+                "Profile name (Nickname): "
+            );
+
+            let config = jenkins.getConfig();
+            let profiles = config.profiles;
+
+            if (profiles === null || profiles === undefined) {
+                profiles = Object.create(null);
+            }
+            profiles[profileName] = jenkinsJobName;
+            config.profiles = profiles;
+
+            jenkins.writeToConfig(config);
+            shell.exit(0);
+        },
+
+        async build() {
+            const userPrompt = prompt.getPrompt();
+            let profile = program.profile;
+            if (StringUtils.isEmpty(program.profile)) {
+                echo.error("Please specify a profile [--profile] to trigger");
+                shell.exit(1);
+            }
+            const config = jenkins.getConfig();
+            if (config.profiles.indexof(profile) < 0) {
+                echo.error(
+                    "Profile not found, please check your configuration"
+                );
+                shell.exit(1);
+            }
+
+            job = config.profiles[profile];
+            tag = await userPrompt.questionAsync("Tag name to deploy: ");
+            this.validateOrExit();
+
+            await prompt.confirmOrExit(
+                `Please confirm deployment of tag ${tag} to ${job}`,
+                0
+            );
+
+            echo.message(`Jenkins starting ${job}, deploying version ${tag}`);
+            const url = `${config.url}/job/${job}/buildWithParameters?version=${tag}`;
+
+            const cmd = commandStringFactory.build(
+                "curl",
+                "-X",
+                "POST",
+                "-L",
+                `--user ${config.username}:${config.token}`,
+                url
+            );
+
+            if (shell.exec(cmd, { silent: false }).code !== 0) {
```

Comment:
Shouldn't need to specify `{ silent: false }` here either, I believe that's the default. Probably copied/pasted from somewhere